api-participant/src/main/java/bio/terra/pearl/api/participant/controller/B2cEventController.java

```diff
@@ -37,14 +35,14 @@ public B2cEventController(B2cEventExtService b2cEventExtService) {
         "https://juniperatcp.b2clogin.com", // ATCP (prod)
         "https://trccproject.b2clogin.com" // tRCC (prod)
       },
-      maxAge = 3600,
-      methods = {RequestMethod.GET, RequestMethod.OPTIONS})
+      allowedHeaders = "*",
+      maxAge = 3600)
   public ResponseEntity<Void> trackFailedLogin(
-      String portalShortcode, String envName, ExternalEventFailedLoginBody body) {
+      String portalShortcode, String envName, String email) {
     try {
       // track asynchronously; caller doesn't care if it succeeds or fails
       b2cEventExtService.trackFailedLoginEventAsync(
-          portalShortcode, EnvironmentName.valueOfCaseInsensitive(envName), body.getUsername());
+          portalShortcode, EnvironmentName.valueOfCaseInsensitive(envName), email);
```

Comment:
separately: there was a CORS issue with the failed login tracking endpoint.

I _think_ it has something to do with the fact that the preflight response was not set up properly. Preflight requests are only sent when the request is considered "complex" by the browser. GET requests are never complex (unless they have unknown headers) and POST requests are not complex IF they use certain content-types. Using application/json is considered complex, but, text/plain isn't.

once I changed the endpoint to just take the email directly instead of using JSON, the cors issue magically disappeared.